ecs_pod_creator.go

```diff
@@ -867,6 +882,61 @@ func (s *SecretOptions) hash() string {
 	return h.Sum()
 }
 
+// LogConfiguration represents the configuration for a container's logging.
+type LogConfiguration struct {
+	// LogDriver is the logging driver to use.
+	LogDriver *string
+	// Options are the logging driver options.
+	Options map[string]*string
+}
+
+// NewLogConfiguration returns a new uninitialized log configuration.
+func NewLogConfiguration() *LogConfiguration {
+	return &LogConfiguration{}
+}
+
+// SetLogDriver sets the logging driver to use.
+func (c *LogConfiguration) SetLogDriver(logDriver string) *LogConfiguration {
+	c.LogDriver = &logDriver
+	return c
+}
+
+// SetOptions sets the logging driver options.
+func (c *LogConfiguration) SetOptions(options map[string]*string) *LogConfiguration {
+	c.Options = options
+	return c
+}
+
+// Validate check that the log driver is set.
+func (c *LogConfiguration) Validate() error {
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(c.LogDriver == nil, "must specify a log driver")
+	catcher.NewWhen(c.Options == nil, "must specify log driver options")
+	if c.Options != nil {
+		_, groupDefined := c.Options["awslogs-group"]
+		_, regionDefined := c.Options["awslogs-region"]
+		catcher.NewWhen(!groupDefined, "must specify awslogs-group in options")
+		catcher.NewWhen(!regionDefined, "must specify awslogs-region in options")
+	}
+	catcher.NewWhen(c.Options != nil && c.Options["awslogs-group"] == nil, "must specify log driver options")
+	return catcher.Resolve()
+}
+
+// hash returns the hash digest of the log configuration.
+func (c *LogConfiguration) hash() string {
+	h := utility.NewSHA1Hash()
+	if c.LogDriver != nil {
+		h.Add(utility.FromStringPtr(c.LogDriver))
+	}
+	if c.Options != nil {
+		for key, value := range c.Options {
```

Comment:
This needs a predictable map iteration order here (like `hashableTagPairs`), or else the hash computed will be different with each iteration on `c.Options`, even for the same map.

Reply:
Sure. Since `tagPair` is just a string key and value, I'll repurpose the existing structs we have for this.